### ISSUE TITLE
Problem with local set up for contributor

### ISSUE THREAD

**Issue by LaPetiteSouris** [kserve/kserve#2903]
/kind bug

**What steps did you take and what happened:**

I decided to spend some time helping out with this amazing project. I followed the [guide set up for new contributors](https://kserve.github.io/website/master/developer/developer/#install-knative-on-a-kubernetes-cluster)

The command `make deploy-dev` failed unexpectedly.

```
Resource: "apiextensions.k8s.io/v1, Resource=customresourcedefinitions", GroupVersionKind: "apiextensions.k8s.io/v1, Kind=CustomResourceDefinition"
Name: "servingruntimes.serving.kserve.io", Namespace: ""
for: "STDIN": error when patching "STDIN": CustomResourceDefinition.apiextensions.k8s.io "servingruntimes.serving.kserve.io" is invalid: spec.validation.openAPIV3Schema.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
make: *** [deploy-dev] Error 1
```
I suspect that it is related to the version mismatches, thus I followed the [guide for versions](https://kserve.github.io/website/0.9/admin/kubernetes_deployment/#recommended-version-matrix)

Am I missing something ? 

**Environment:**

- Istio Version:
    ```
    client version: 1.13.8
    control plane version: 1.13.8
    data plane version: none
    
    ```
- Knative Version:
    ```
    Name:         knative-eventing
    Labels:       app.kubernetes.io/name=knative-eventing
                  app.kubernetes.io/version=1.6.0
                  eventing.knative.dev/release=v1.6.0
                  kubernetes.io/metadata.name=knative-eventing
    
    
    Name:         knative-serving
    Labels:       app.kubernetes.io/name=knative-serving
                  app.kubernetes.io/version=1.6.0
                  kubernetes.io/metadata.name=knative-serving
    
    ```
- KServe Version: local fork, identical to `master` branch

- Kubeflow version:
- Cloud Environment: None
- Minikube/Kind version: `minikube version: v1.30.1`
- Kubernetes version: (use `kubectl version`): `1.23`
- OS (e.g. from `/etc/os-release`): MacOS M1 Sillicon `13.3.1`
- go version: `go1.19.9 darwin/arm64`

**Comment by sivanantha321** [comment]
What version of kustomize you are using? Try with kustomize 5.0

**Comment by LaPetiteSouris** [comment]
> What version of kustomize you are using? Try with kustomize 5.0

Thanks. I was using `kustomize v5.0.3` and downgraded to `5.0.0` as suggested, however, the same error persists. It looks likely this error is related to some versions, it is just I don't know which to try.

I may spawn up a cheap throw-away K8s 1.23 cluster to try, but not sure if it helps as everything should work on either `kind` or `minikube`. I tried both.

**Comment by sivanantha321** [comment]
Maybe you can try Installing kserve using the [quickstart script](https://github.com/kserve/kserve/blob/master/hack/quick_install.sh). Then try `make deploy-dev`.

**Comment by LaPetiteSouris** [comment]
Thanks. The quickstart script manages to deploy the pre-build kserve on `minikube`. However, afterward, I still get the same error on `make deploy-dev` and `make deploy`. Tested on 2 separate environment (one is Apple Silicon CPU and another is AMD64 CPU). Same error.

Any thoughts ?

**Comment by LaPetiteSouris** [comment]
I found https://github.com/kserve/kserve/issues/977 to be similar, however, I think since then `controller-gen` is pinned already in `go.mod` so I think it is irrelevant now. Quite strange...

Also, seems like there is someone else mentioned this same error before at https://github.com/kserve/kserve/issues/2422#issuecomment-1246519609

**Comment by sivanantha321** [comment]
can you try with these:
  
controller-gen version: 0.4.0
kubectl version: 1.26.0
ko version:  0.13.0
yq version :  4.28.1
kustomize version: 5.0.1

**Comment by LaPetiteSouris** [comment]
Thanks. I just found the problem. I set up all dependencies as suggested by @sivanantha321 , which still resulted in the same error. This led me  triangle it to `minikube` or `knative` or `istio`(the only 3 things left that have not been tried).

Not sure how it could affect, as I do not understand things well enough. But here is the issue:

- When installing `Knative`, I used the [knative quickstart](https://knative.dev/docs/install/quickstart-install/) provided by `knative` too bootstrap the stack. This problem happens when `knative quickstart` is used.
- If I install `knative` and `istio` [using the standard YAML](https://knative.dev/docs/install/yaml-install/serving/install-serving-with-yaml/), `make deploy-dev` works smoothly and I can run tests locally on minikube

I will try to drop a PR on the docs to add this note, just to save time for our next contributors.
